packages/timeline-state-resolver/src/service/device.ts

```diff
@@ -148,6 +152,9 @@ export interface DeviceContextAPI {
 
 	timeTrace: (trace: FinishedTrace) => void
 
-	/** Reset the state of the State */
+	/** Reset the tracked device state to undefined and notify the conductor to reset the resolver */
 	resetState: () => Promise<void>
+
+	/** Reset the tracked device state to "state" and notify the conductor to reset the resolver */
+	resetToState: (state: any) => Promise<void> // todo - types?
```

Comment:
Unresolved TODO
I agree it would be good to have the type of the state to be known here. It will probably be a bit sprawling, but should be easy enough to do as a generic parameter to this interface

Reply:
My initial worry was that we already use a lot of generics and it does become a bit of a sprawl but in this case it's only a few places that need to be aware actually.